src/lib/xwalk-hack/churn.ts

```diff
@@ -0,0 +1,205 @@
+/// <reference path='../../../../third_party/ipaddrjs/ipaddrjs.d.ts' />
+/// <reference path='../../../../third_party/typings/browser.d.ts' />
+
+import caesar = require('../transformers/caesar');
+import candidate = require('../churn/candidate');
+import churn_types = require('../churn/churn.types');
+import handler = require('../handler/queue');
+import logging = require('../logging/logging');
+import peerconnection = require('../webrtc/peerconnection');
+import random = require('random-lib');
+import signals = require('../webrtc/signals');
+
+import Candidate = candidate.Candidate;
+
+import ChurnSignallingMessage = churn_types.ChurnSignallingMessage;
+import TransformerConfig = churn_types.TransformerConfig;
+
+var log :logging.Log = new logging.Log('xchurn');
+
+export var getMidFromSdp = (sdp:string) : string => {
```

Comment:
I'm guessing this code isn't 100% review ready - but FYI it doesn't look like this is called?